bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/api/model/PlugwiseHAControllerRequest.java

```diff
@@ -225,6 +227,22 @@ private ContentResponse getContentResponse(int retries) throws PlugwiseHAExcepti
 
         this.logger.debug("Performing API request: {} {}", request.getMethod(), request.getURI());
 
+        if (logger.isTraceEnabled()) {
+            String content = "";
+            final ContentProvider provider = request.getContent();
+            if (provider != null) {
+                final Iterator<ByteBuffer> it = provider.iterator();
+                while (it.hasNext()) {
+                    final ByteBuffer next = it.next();
+                    final byte[] bytes = new byte[next.capacity()];
+                    next.get(bytes);
+                    content = new String(bytes, StandardCharsets.UTF_8);
```

Comment:
That us strange, you are looping but finally consider only the last item.

Reply:
Not intentionally. This remote API always responds resulting with 1 item in the iterator. I changed it so that if it holds more it is stored to the output. Thanks, good catch!